### PR TITLE
Fix UI hashgraph sorting bug.

### LINKED ISSUE #6952: Fix event sorting in hashgraph UI
For the longest time, there has been a race condition in the hashgraph UI used for local testing. Fix that bug.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #6952

### CODE REVIEW

**Review by edward-swirldslabs — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java [hunk 1]
```diff
@@ -185,40 +182,68 @@ public ShadowGraph getShadowGraph(@NonNull NodeId nodeId) {
...
-                logger.error(EXCEPTION.getMarker(), "Exception while sorting events", e);
-            }
+
+        final EventImpl[] allEvents = getShadowGraph(nodeId).getAllEvents();
+
+        // If events reach consensus during the execution of this method, order will change.
+        // Capture information before sorting, since sorting breaks if order changes mid-sort.
+        final Map<Hash, EventOrderInfo> orderInfo = new HashMap<>();
+        for (final EventImpl event : allEvents) {
+            orderInfo.put(event.getHashedData().getHash(), EventOrderInfo.of(event));
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java:236]: Are you intentionally using `getHashedData` here, but `getBaseEventHashedData` below?

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java:236]: These two methods are aliases for the same thing. No idea why there are two completely different methods, but not really in the scope of this PR to remove one of them. I've changed my new code to only use `getHashedData()`, so at least this part of the code is self consistent.

**Comment by cody-littley** [comment]
@lpetrovic05 

>why are we not just sorting these events by generation?

No idea, I'm just carrying forward the previous logic in a way that isn't going to throw exceptions.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java [hunk 2]
```diff
@@ -185,40 +182,66 @@ public ShadowGraph getShadowGraph(@NonNull NodeId nodeId) {
...
 
+    /**
+     * Information required to sort events by consensus order (or, if events have not yet reached consensus, by received
+     * time).
+     *
+     * @param consensusOrder the consensus order if known, else -1 if unknown
+     * @param timeReceived   the time the event was received, used to break ties when consensus order is unknown for
+     *                       both events
+     */
+    private record EventOrderInfo(long consensusOrder, @NonNull Instant timeReceived)
```
Comment (qnswirlds) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java:193]: nit:
```suggestion
    private record EventOrderInfo(final long consensusOrder, @NonNull final Instant timeReceived)
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java [hunk 3]
```diff
@@ -185,40 +182,66 @@ public ShadowGraph getShadowGraph(@NonNull NodeId nodeId) {
...
+                // both have reached consensus
+                return Long.compare(this.consensusOrder, that.consensusOrder);
+            }
+        }
+    }
+
     /**
      * Get a sorted list of events.
      */
     public PlatformEvent[] getAllEvents(@NonNull NodeId nodeId) {
```
Comment (qnswirlds) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java:227]:
```suggestion
    public PlatformEvent[] getAllEvents(@NonNull final NodeId nodeId) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java:193]: records don't use `final` (enforced by compiler)

**Pull request merged**